### ISSUE TITLE
Item not imputed in batch run

### ISSUE THREAD

**Issue by mkao006** [SWS-Methodology/faoswsProduction#196]
There appear to be some item not imputed when running on the complete production data but imputed when running on session.

We suspect it is because the item is not in the complete imputation key specified. But when the session is specified, it can impute anything.

Now we need to check whether pistachio (01375) is in the complete imputation key.

**Comment by mkao006** [comment]
Also check the list provided by Tomasz, I suspect the oils are not in the list as well.

**Comment by mkao006** [comment]
Pistachio is in the list and oil as well. Need to investigate further why they are not imputed or saved back.

**Comment by mkao006** [comment]
It seems the error comes from the values are protected in the database, and thus can not be over-written.

"Error in SaveData(domain = sessionKey@domain, dataset = sessionKey@dataset,  : 
  Validation error/s occurred:
[1757, 65]: Cannot overwrite an official data item
[1759, 65]: Cannot overwrite an official data item
[1761, 65]: Cannot overwrite an official data item
[1842, 65]: Cannot overwrite an official data item

However, they are calculated official data (, i), and thus should be writable. Awaiting response from Giorgio.

**Comment by mkao006** [comment]
After a discussion with Giorgio, we have uncovered why the R module was unsuccessful in saving the imputed values back to the system.

The reason was due to the protection mechanism of the system preventing R modules over-writting official values and interfere with other data collection procedures such as the harvestor.

Thus, despite the execution being a core module, the R module does not have the privilege to over-write official values.

The solution is to run the imputation in session then manually save the data back to the database.
​
